include/oneapi/dpl/internal/dynamic_selection_impl/static_policy_impl.h

```diff
@@ -0,0 +1,117 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _STATIC_POLICY_IMPL_H
+#define _STATIC_POLICY_IMPL_H
+
+#pragma once
+#include <ostream>
+//#include "oneapi/dpl/internal/dynamic_selection.h"
+#include "oneapi/dpl/internal/dynamic_selection_impl/scoring_policy_defs.h"
+
+namespace oneapi {
+namespace dpl {
+namespace experimental {
+
+  template <typename Scheduler>
+  struct static_policy_impl {
+    using scheduler_t = Scheduler;
+    using native_resource_t = typename scheduler_t::native_resource_t;
+    using execution_resource_t = typename scheduler_t::execution_resource_t;
+    using native_sync_t = typename scheduler_t::native_sync_t;
+    using universe_container_t = typename scheduler_t::universe_container_t;
+    using selection_handle_t = oneapi::dpl::experimental::nop_selection_handle_t<execution_resource_t>;
+
+    std::shared_ptr<scheduler_t> sched_;
+    universe_container_t universe_;
+
+    static_policy_impl() : sched_{std::make_shared<scheduler_t>()} {
+      universe_ = oneapi::dpl::experimental::property::query(*sched_, oneapi::dpl::experimental::property::universe);
+    }
+
+    static_policy_impl(universe_container_t u) : sched_{std::make_shared<scheduler_t>()} {
+      oneapi::dpl::experimental::property::report(*sched_, oneapi::dpl::experimental::property::universe, u);
+      universe_ = oneapi::dpl::experimental::property::query(*sched_, oneapi::dpl::experimental::property::universe);
+    }
+
+    template<typename ...Args>
+    static_policy_impl(Args&&... args) : sched_{std::make_shared<scheduler_t>(std::forward<Args>(args)...)} {
+      universe_ = oneapi::dpl::experimental::property::query(*sched_, oneapi::dpl::experimental::property::universe);
+    }
+
+    //
+    // Support for property queries
+    //
+
+    auto query(oneapi::dpl::experimental::property::universe_t) const noexcept {
+      return oneapi::dpl::experimental::property::query(*sched_, oneapi::dpl::experimental::property::universe);
+    }
+
+    auto query(oneapi::dpl::experimental::property::universe_size_t) const noexcept {
+      return oneapi::dpl::experimental::property::query(*sched_, oneapi::dpl::experimental::property::universe_size);
+    }
+
+    template<typename ...Args>
+    selection_handle_t select(Args&&...) {
```

Comment:
Is there any policy we really need those `Args&&...`

Reply:
In the future, but not in beta. Something like auto-tuning might want the "key" to be more than just the lambda; i.e. the lambda plus some argument values, such as data size.